cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -170,16 +191,22 @@ private boolean doublyEscaped(int index, String str) {
   }
 
   /**
-   * Removes all escaped syntax for all macro syntax symbols: ${, }, (, )
-   * TODO: Handle doubly-escaped syntax (e.g. \\${macro} should still be evaluated)
+   * Removes all backslash '\' characters unless they are preceded by a backslash.
```

Comment:
this is not exactly accurate, as '\\' will be replaced with '\'. I think you can just say that it strips preceding '\'.